engine/src/main/java/org/terasology/rendering/dag/nodes/OverlaysNode.java

```diff
@@ -19,25 +19,50 @@
 import org.terasology.entitySystem.systems.RenderSystem;
 import org.terasology.monitoring.PerformanceMonitor;
 import org.terasology.registry.In;
+import org.terasology.rendering.cameras.Camera;
 import org.terasology.rendering.dag.WireframeCapableNode;
+import org.terasology.rendering.dag.stateChanges.SetViewportToSizeOf;
+import org.terasology.rendering.world.WorldRenderer;
+
+import static org.terasology.rendering.opengl.DefaultDynamicFBOs.READ_ONLY_GBUFFER;
 
 /**
- * TODO: Diagram of this node
+ * This nodes renders overlays, i.e. the black lines highlighting a nearby block the user can interact with.
+ *
+ * Objects to be rendered as overlays must be registered as implementing the interface RenderSystem and
+ * must take advantage of the RenderSystem.renderOverlay() method, which is called in process().
  */
 public class OverlaysNode extends WireframeCapableNode {
 
     @In
     private ComponentSystemManager componentSystemManager;
 
+    @In
+    private WorldRenderer worldRenderer;
+
+    private Camera playerCamera;
+
+    /**
+     * Initialises the node. -Must- be called once after instantiation.
+     */
     @Override
     public void initialise() {
         super.initialise();
+        playerCamera = worldRenderer.getActiveCamera();
+        addDesiredStateChange(new SetViewportToSizeOf(READ_ONLY_GBUFFER));
     }
 
+    /**
+     * Iterates over any registered RenderSystem instance and calls its renderOverlay() method.
+     */
     @Override
     public void process() {
         PerformanceMonitor.startActivity("rendering/overlays");
 
+        playerCamera.lookThrough(); // TODO: remove. Placed here to make the dependency explicit.
```

Comment:
All nodes are dependent on a camera from which something is rendered. Previously these nodes relied on an identical line placed upstream in a previous node. If a second camera was set between that node and this node, this node would render through the second camera instead. 

By placing this line in this and other nodes I make the dependency on this particular camera explicit and these nodes can move in the render graph slightly more freely. 

Eventually this line (and the identical ones in other nodes) will be replaced by a StateChange and/or a setCamera() method. In turn these ways to set the camera for the node will eventually take in input an entity with an attached camera component rather than an instance of a camera class. But that's still some way out.